### PR TITLE
Testing smart contracts for wiki

### PR DESCRIPTION
**Pull request by AndrePanin**

### CODE REVIEW

File: docs/developing-contracts/messaging.md [hunk 1]
```diff
@@ -55,24 +58,22 @@ pub fn load_bytes() -> Vec<u8> {
     result
 }
 ```
-Load bytes
+
+Reply to a message and try to decode to the specified type with `SCALE codec`. Returns `MessageId`:
```
Comment (breathx) [inline docs/developing-contracts/messaging.md:62]: This one encodes the argument and sends it. No decode functions

File: docs/developing-contracts/testing.md [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+    [build]
+    target = "wasm32-unknown-unknown"
+
+    [target.wasm32-unknown-unknown]
+    rustflags = ["-C", "link-args=--import-memory"]
+    ```
+    In this case, you have to choose the most convenient option for you:
+
+    * (Recommended) Remove build target `target=wasm32-unknown-unknown` from `config.toml`, which means that you have to
+        * Build the contact with: `cargo +nightly build --target wasm32-unknown-unknown`
```
Comment (breathx) [inline docs/developing-contracts/testing.md:55]: Contract*
Typo

File: docs/developing-contracts/testing.md [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+        * Run test as: `cargo test` (опционально: `cargo +nightly test`)
```
Comment (breathx) [inline docs/developing-contracts/testing.md:56]: Russian language 😖

File: docs/developing-contracts/testing.md [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+
+    * Keep target in the configuration file, in this case:
+        * Build the contact with: `cargo +nightly build`
```
Comment (breathx) [inline docs/developing-contracts/testing.md:59]: Same typo

File: docs/developing-contracts/testing.md [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+        * Run test as: `cargo test --target any_not_wasm32_unknown_unknown_target`
+
+4. Every supported by rust target available from this command:
```
Comment (breathx) [inline docs/developing-contracts/testing.md:62]: Probably Rust with capital letter

File: docs/developing-contracts/testing.md [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+use gtest::{Log, Program, System};
+
+#[test]
+fn basics() {
+    // Initialization of the common environment for running smart contacts.
+    //
+    // This emulates node's and chain behavior.
+    //
+    // By default, sets:
+    // - current block equal 0
```
Comment (breathx) [inline docs/developing-contracts/testing.md:126]: Equals

File: docs/developing-contracts/testing.md [hunk 7]
```diff
@@ -0,0 +1,298 @@
...
+    // Initialization of program structure from file.
+    //
+    // Takes as arguments reference to the related `System` and the path to wasm binary relatively 
+    // the root of the crate where the test was written.
+    //
+    // Sets free program id from the related `System` to this program. For this case it equals 0x010000..
+    // Next program initialized without id specification will have id 0x020000.. and so on.
+    let ping_pong = Program::from_file(
+        &sys,
+        "../target/wasm32-unknown-unknown/release/demo_ping.wasm",
```
Comment (breathx) [inline docs/developing-contracts/testing.md:155]: Replace everywhere "../target" with "./target", my fault due to workspace in apps repo

**Review by breathx — APPROVED** [review]

File: docs/developing-contracts/messaging.md (hunk 1 above)
Comment (AndrePanin) [reply docs/developing-contracts/messaging.md]: done

File: docs/developing-contracts/testing.md (hunk 2 above)
Comment (AndrePanin) [reply docs/developing-contracts/testing.md]: done

File: docs/developing-contracts/testing.md (hunk 3 above)
Comment (AndrePanin) [reply docs/developing-contracts/testing.md]: done

File: docs/developing-contracts/testing.md (hunk 4 above)
Comment (AndrePanin) [reply docs/developing-contracts/testing.md]: done

File: docs/developing-contracts/testing.md (hunk 5 above)
Comment (AndrePanin) [reply docs/developing-contracts/testing.md]: done

File: docs/developing-contracts/testing.md (hunk 6 above)
Comment (AndrePanin) [reply docs/developing-contracts/testing.md]: done

File: docs/developing-contracts/testing.md (hunk 7 above)
Comment (AndrePanin) [reply docs/developing-contracts/testing.md]: done

**Pull request merged**